src/analyses/local_safe_pointers.cpp

```diff
@@ -0,0 +1,239 @@
+/*******************************************************************\
+
+Module: Local safe pointer analysis
+
+Author: Diffblue Ltd
+
+\*******************************************************************/
+
+/// \file
+/// Local safe pointer analysis
+
+#include "local_safe_pointers.h"
+
+#include <langapi/language_util.h>
+#include <util/expr_iterator.h>
+
+/// If `expr` is of the form `x != nullptr`, return x. Otherwise return nil.
+static exprt get_null_checked_expr(const exprt &expr)
```

Comment:
You really want to use pointers in the below code and have `const exprt &` as return type.